local-env-setup.sh

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+darwin_install_pip3_packages() {
+    echo "Installing setuptools grpcio-tools virtualenv"
+    pip3 install setuptools grpcio-tools virtualenv
+    echo "Installing mypy-protobuf"
+    pip3 install --user mypy-protobuf
+}
+
+install_go_packages(){
+        echo "Installing goavro"
+        go get github.com/linkedin/goavro
+        # As we are using bash, we are assuming .bashrc exists. 
+        grep -qxF "export GOPATH='${pwd}/sdks/go/examples/.gogradle/project_gopath'" ~/.bashrc || echo "export GOPATH='${pwd}/sdks/go/examples/.gogradle/project_gopath'" >> ~/.bashrc
```

Comment:
Is it possible to set `GOPATH` in this script, but not write it to `.bashrc`? I'm a little worried about messing around with `.bashrc` files automatically.

We could,

1. check if GOPATH is set already
2. if not, set it locally in this script, print out a line to the user telling them to add it to their bashrc file


@lostluck could you review the go related config please?

Reply:
Changed the logic to set the `GOPATH` variable only for the script session if it's not found in `.bashrc ` and added an `echo` to remind the user to add it manually.